### PR TITLE
fix(sql/tsql): add support for GROUPING SETS

### PR DESCRIPTION
**Pull request by grzeng-go**
Related Documentation:
 https://docs.microsoft.com/en-us/sql/t-sql/queries/select-group-by-transact-sql?view=sql-server-ver15

#2590

### CODE REVIEW

File: sql/tsql/TSqlParser.g4 [hunk 1]
```diff
@@ -3278,7 +3278,7 @@ query_specification
       (FROM from=table_sources)?
       (WHERE where=search_condition)?
       // https://msdn.microsoft.com/en-us/library/ms177673.aspx
-      (GROUP BY groupByAll=ALL? groupBys+=group_by_item (',' groupBys+=group_by_item)*)?
+      (GROUP BY ((groupByAll=ALL? groupBys+=group_by_item (',' groupBys+=group_by_item)*) | GROUPING SETS '(' groupSets+=grouping_sets_item (',' groupSets+=grouping_sets_item)* ')'))?
```
Comment (KvanTTT) [inline sql/tsql/TSqlParser.g4:3281]: It seems like there is the excess pair of parentheses:

```suggestion
      (GROUP BY (groupByAll=ALL? groupBys+=group_by_item (',' groupBys+=group_by_item)* | GROUPING SETS '(' groupSets+=grouping_sets_item (',' groupSets+=grouping_sets_item)* ')'))?
```

Comment (grzeng-go) [reply sql/tsql/TSqlParser.g4:3281]: cannot be removed, because if removed, GROUPING will be treated as group_by_item

Comment (KvanTTT) [reply sql/tsql/TSqlParser.g4:3281]: `|` has lower priority than concatentation.

Comment (grzeng-go) [reply sql/tsql/TSqlParser.g4:3281]: eg: SELECT * FROM A GROUP BY GROUPING SETS (())
 if removed, GROUPING will be identified as group_by_item,
I tested it with the antlr4 plugin and it is like this

**Pull request merged**